src/main/scala/atom/Atom.scala

```diff
@@ -0,0 +1,113 @@
+package laughedelic.atom
+
+import scala.scalajs.js, js.annotation._
+
+@js.native
+trait Disposable extends js.Object {
+  def dispose(): Unit = js.native
+}
+
+@js.native
+trait CommandRegistry extends js.Object {
+  def add(target: String, commandName: String, callback: js.UndefOr[js.Any => js.Any]): Unit = js.native
+  def add(target: String, commands: js.Dictionary[js.Any => js.Any], callback: js.UndefOr[js.Any => js.Any]): Unit = js.native
+}
+
+@js.native
+trait Model extends js.Object {
+  def destroy(): Unit = js.native
+  def isDestroyed(): Boolean = js.native
+}
+
+
+@js.native
+trait TextEditor extends Model {
+  val id: Long = js.native
+  
+  val firstVisibleScreenRow: Long = js.native
+  val rowsPerPage: Long = js.native
+
+  // File Details
+  def getTitle(): String = js.native
+  def getLongTitle(): String = js.native
+
+
+  def getPath(): js.UndefOr[String] = js.native
+  def getURI(): js.UndefOr[String] = js.native
+  def insertNewline(): Unit = js.native
+  def isModified(): Boolean = js.native
+  def isEmpty(): Boolean = js.native
+  // def getEncoding(): Encoding = js.native
+  def setEncoding(encoding: String): Unit = js.native
+  def getTabLength(): Long = js.native
+
+  // File Operations
+  def save(): Unit = js.native
+  def saveAs(filePath: String): Unit = js.native
+
+  // Reading Text
+  def getText(): String = js.native
+  def getLineCount(): Long = js.native
+}
+  
+@js.native
+trait ConfigChange extends js.Object {
+  val newValue: js.Any = js.native
+  val oldValue: js.UndefOr[js.Any] = js.native
+}
+
+@js.native
+trait Config extends js.Object {
+  
+  def observe(keyPath: String, callback: ConfigChange => Unit): Disposable = js.native
+  def onDidChange(callback: ConfigChange => Unit): Disposable = js.native
+  def onDidChange(keyPath: String, callback: ConfigChange => Unit): Disposable = js.native
+  
+  def get(key: String): js.Any = js.native
+  def set(key: String, value: js.UndefOr[js.Any] = js.undefined): Boolean = js.native
+  def unset(key: String): Boolean = js.native
```

Comment:
These methods have some `options` optional argument. It's probably not extremely useful, but I suggest either adding a facade for it for completeness, or just a TODO comment saying that something is missing here. This way it's easier to see later which parts of the original API are missing and could be improved.